lib/libtlspool.c

```diff
@@ -37,6 +38,18 @@ static pthread_cond_t updated_poolfd = PTHREAD_COND_INITIALIZER;
 
 static pthread_mutex_t prng_lock = PTHREAD_MUTEX_INITIALIZER;
 
+int tlspool_getpid () {
+	int fd;
+
+	if ((fd = open("/var/run/tlspool.pid", O_RDONLY)) != -1) {
+		char str_pid[256];
+
+		if (read(fd, str_pid, sizeof(str_pid)) != -1) {
```

Comment:
Your buffer does not terminate... and you are not validating the PID file contents... suggested code:

#include <climits.h>
#include <ctypes.h>

char *endptr;
size_t len;

len = read (fd, str_pid, sizeof (str_pid) -1);
if (len > 0) {
   str_pid [len] = '\0;
   errno = 0;
   unsigned long pid = strtoul (str_pid, &endptr, 10);
   if ((errno == 0) && (endptr != NULL) && (isspace (endptr)) && (pid <= INT_MAX)) {
      return (int) pid;
   }
   return -1;
}